src/lib.rs

```diff
@@ -1,8 +1,230 @@
 // Copyright 2019-2023 ChainSafe Systems
 // SPDX-License-Identifier: Apache-2.0, MIT
 
+//! # Actors
+//!
+//! The Filecoin Virtual Machine (FVM) hosts a small[^1] number of _actors_.
+//! These are objects that maintain and mutate internal state, and communicate
+//! by passing messages.
+//!
+//! An example of an actor is the [`cron`](fil_actors_shared::v11::runtime::builtins::Type::Cron)
+//! actor.
+//! Its [internal state](fil_actor_cron_state::v11::State) is a to-do list of
+//! other actors to invoke every epoch.
+//!
+//! See [the Filecoin docs](https://docs.filecoin.io/basics/the-blockchain/actors)
+//! for more information about actors.
+//!
+//! [^1]: The number of [built-in actors](https://docs.filecoin.io/basics/the-blockchain/actors#built-in-actors)
+//!       is small.
+//!
+//! TODO(aatifsyed): where is my balance?
+//!
+//! # The Filecoin blockchain
+//!
+//! Filecoin consists of a blockchain of messages.
+//! Listed below are the core objects for the blockchain.
+//! Each one can be addressed by a [`Cid`](cid::Cid).
+//!
+//! - [`Message`](shim::message::Message)s are statements of messages between
+//!   a the actors.
+//!   They describe and (equivalently) represent a change in _the blockchain state_.
+//!   See [`apply_block_messages`](state_manager::apply_block_messages) to learn
+//!   more.
+//!   Messages may be [signed](message::SignedMessage).
+//!   TODO(aatifsyed): by whom? what does that mean?
```

Comment:
From https://spec.filecoin.io/systems/filecoin_vm/message/ :

"A semantically valid `SignedMessage` must carry a signature that verifies the payload as having been signed with the public key of the account actor identified by the `From` address. Note that when the `From` address is an ID-address, the public key must be looked up in the state of the sending account actor in the parent state identified by the block."